src/QMCDrivers/DMC/WalkerControlMPI.cpp

```diff
@@ -390,4 +447,163 @@ void WalkerControlMPI::swapWalkersSimple(MCWalkerConfiguration& W)
   }
 }
 
+/** swap Walkers between rank MCPopulations
+ *
+ *  2005 called and asked for its synchronous messaging back
+ *  I think MPI async is more than two years old
+ *
+ *  MCPopulation is sufficiently different from MCWalkerConfiguration that this is 
+ *  basically a rewrite.  I have not replicated the more ridiculous elements of the original.
+ *  For example:
+ *   * Sending duplicates to the same reciever rank but resorting good walkers if send-receive pair 
+ *     changes.
+ *   * Sending a tiny separate blocking message to tell the receiver how many copies
```

Comment:
This is intended to avoid sending duplicates to the same receiver. The two items mentioned here are contradictory to each other.